### PR TITLE
[13.0] Add edi

### PR DESCRIPTION
**Pull request by simahawk**
Depends on #247 
To be considered for #243.
To be the base for more work to be discussed w/ @etobella

### CODE REVIEW

**Comment by simahawk** [comment]
@etobella I renamed the module to `edi` as we dicussed :wink: 

I'll rebase after #247 is merged.

**Comment by etobella** [comment]
You can rebase now :smile:

File: edi/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+{
+    "name": "Base EDI Backend",
+    "summary": """Base module to define EDI backends""",
+    "version": "13.0.1.0.0",
+    "development_status": "Alpha",
+    "license": "AGPL-3",
+    "author": "ACSONE,Odoo Community Association (OCA)",
+    "maintainers": ["simahawk"],
+    "depends": ["base_edi", "connector"],
```
Comment (bealdav) [inline edi/__manifest__.py]: Thanks for this module.
Just a question here. Do you think that connector is mandatory in this module or only component/queue_job ? Maybe connector in an additional module ?

Comment (etobella) [reply edi/__manifest__.py]: At least component is required (with the changes we are working on), and probably connector will be required.

Comment (simahawk) [reply edi/__manifest__.py]: yeah, for now we can stick to component only indeed. Fixed.

**Comment by simahawk** [comment]
@bealdav @etobella good for you?

**Comment by etobella** [comment]
@simahawk I made a PR on your branch that adds most of the functionality we talked about:

https://github.com/simahawk/edi/pull/1

Taking all the points you marked on #249 I think I implemented some of them, and the others, might be easy to add as the idea is there, but I would like to talk about them in order to listen all opinions:

- [x] have a generic EDI backend classified by type (#248)
- [x] describe exchanges by type or format (eg: outgoing vs incoming, needs a file, needs an ack, etc)
- [x] track exchanges w/ a generic record (either edi.exchange or edi.document)
- [ ] have a component mechanism to customize generation or parsing of data
- [ ] have a component mechanism to talk to external protocols (using storage modules or api calls)
- [ ] have generic hooks and events to customize what happens when records are received or sent --> We might need `component_event`, isn't it?
- [ ] have a way to correlate exchanged records --> I think this might be managed with the model, res_id fields, so not sure if it is necessary

This point might be going to a new module:

- have a templating system to define generation or parsing of data

**Comment by simahawk** [comment]
@etobella thanks!
If this base is ok for you then let's merge this little bit here. Meanwhile I review your PR and then we can move it to this repo.
Remember: let's try to work with small chunks :wink:

**Comment by etobella** [comment]
I know, the PR is just all the information together, but probably it would be better to split it in several PRs /Commits in order to review it properly

**Review by etobella — APPROVED** [review]

**Comment by etobella** [comment]
I already splited the PR in several commits that can be checked one by one in different PRs :smile: 

https://github.com/tegin/edi/commits/13.0-edi-components

**Comment by simahawk** [comment]
@etobella exactly what I had in mind, amazing! :smile:

**Pull request merged**